app/controllers/books_controller.rb

```diff
@@ -4,7 +4,13 @@ class BooksController < ApplicationController
   before_action :authenticate_user!
   
   def index
-    render_paginated Book.all, each_serializer: BookSerializer
+    books = Rack::Reducer.call(params, dataset: Book.all, filters: [
```

Comment:
you can use this section in RackReducer documentation https://github.com/chrisfrank/rack-reducer#with-rails-scopes

and define the filter inside the model and in controller only call the apply method